### DISCUSSION TITLE
[Idea] Printer inventory with computers

### DISCUSSION THREAD

**Discussion by Serpher1** [glpi-project/glpi-agent#814] [Ideas]
### Your idea

Will GLPI Agent scan for printers that are connected to a computer (and attach them to said computer) in the future like OCS plugin did?

**Comment by g-bougard** [comment]
Hi @Serpher1 

glpi-agent still tries to inventory printers. Then, in GLPI, printers can be linked to the computer where it has been inventoried.

Anyway, sometimes, depending on the OS, the printer and even the way the printer is connected, you may not obtain what you expect. In that case, you can open an issue and we will see if we can enhance glpi-agent to support your case or help you to figure out how to fix the problem.

If you have a more accurate problem with detailed context, maybe I can provide a more accurate answer. Sorry I don't know what does OCS actually, and so I won't try to compare current GLPI native inventory to what OCS does.

**Reply by Serpher1** [reply]
I noticed that glpi-agent will attach any device that is USB connected but won't read any other printers that are installed and by that I mean IP printers added in Windows. OCS printers plugin (.JS) used to do this.
![2024-12-30 10_54_28](https://github.com/user-attachments/assets/dab243db-670f-4cc6-beab-606ee48ace07)

**Reply by g-bougard** [reply]
On the related computer, can you share the output of the following command (at least the "printers" json node) ? This command must be run from an administrative console and from the glpi-agent folder:
```
glpi-inventory --partial=printer
```

Can you also share the output of the following command as glpi-agent analyses related WMI objects ?
```
wmic path Win32_Printer get /format:list
```
